executor/load_data.go

```diff
@@ -408,39 +408,73 @@ func (e *LoadDataInfo) indexOfTerminator(bs []byte) int {
 	fieldTermLen := len(fieldTerm)
 	lineTerm := []byte(e.LinesInfo.Terminated)
 	lineTermLen := len(lineTerm)
-	length := len(bs)
+	type termType int
+	const (
+		notTerm termType = iota
+		fieldTermType
+		lineTermType
+	)
+	// likely, fieldTermLen should equal to lineTermLen, compare fieldTerm first can avoid useless lineTerm comparison.
+	cmpTerm := func(restLen int, bs []byte) (typ termType) {
+		if restLen >= fieldTermLen && bytes.Equal(bs[:fieldTermLen], fieldTerm) {
+			typ = fieldTermType
+			return
+		}
+		if restLen >= lineTermLen && bytes.Equal(bs[:lineTermLen], lineTerm) {
+			typ = lineTermType
+			return
+		}
+		return
+	}
+	if lineTermLen > fieldTermLen && bytes.HasPrefix(lineTerm, fieldTerm) {
```

Comment:
How about the `line term` and `filed term` are the same?

Reply:
So return the first line term is acceptable? anyway, the user input is misleading...

Reply:
no, current or before issue's behavior is different from mysql....

mysql will insert 2 rows for `|1|2|3|4` into `create table (a int, b int)`...it's very corner case - -